### PR TITLE
refactor: reduce SCSS

### PR DESCRIPTION
**Pull request by Cup0fCoffee**
## Description

Reduces the amount of SCSS in the repo, by applying the styling using bootstrap classes wherever possible.

The main goal is to reduce number of instances where bootstrap variables are used in custom styles, and replace them by applying bootstrap classes directly on the elements. E.g. removing style `color: $gray-300;`, and adding `text-gray-300` class to disabled links in the header.

The secondary goals are to remove unused styles (i.e. styles that don't have any effect) and remove custom styles that can be replaced by bootstrap classes.

### General setup

* Setup local `master` devstack
* Checkout the branch in `frontend-app-publisher`
* Run `make frontend-app-publisher-up studio-up` to spin up necessary containers
* Go to `localhost:18000` and sign in
* Go to `localhost:18400`
* Create a new course

### Testing autosuggest drop down

* From `localhost:18400` navigate to the course you've created
* Open up course information
* Scroll down to `Search or add new collaborator:`
* Start typing until you see a suggestion drop down show up
* Create a new collaborator
* Remove it from the course and make the drop down appear again
* Make sure the drop down looks the same as before the modifications

### Testing header

* Go to `localhost:18400`
* Make sure the header looks the same as before
* Apply the following patch to force the links in the headers to appear disabled:
```diff
diff --git a/src/components/Header/index.jsx b/src/components/Header/index.jsx
index 2cdd111..03a2ab7 100644
--- a/src/components/Header/index.jsx
+++ b/src/components/Header/index.jsx
@@ -24,7 +24,8 @@ const Header = ({ darkModeOn, location, toggleDarkMode }) => {

   function disableLink(courseType) {
     const courseTypeFromURL = querystringParams.course_type ? querystringParams.course_type : '';
-    return (courseTypeFromURL === courseType && pathname === '/') ? 'disabled-link text-gray-300' : '';
+    return 'disabled-link text-gray-300';
+    // return (courseTypeFromURL === courseType && pathname === '/') ? 'disabled-link text-gray-300' : '';
   }

   return (
```
* Compare that disabled links look the same as before

### Gray out text area in disabled RichEditor editor

Couldn't remove custom style in file [`src/components/RichEditor/RichEditor.scss`](https://github.com/open-craft/frontend-app-publisher/blob/40a1e3b3cb23cf271d35c4fbeb14bbd0b662023a/src/components/RichEditor/RichEditor.scss) which is using a boostrap variable:
```scss
.disabled-rich-text .tox .tox-edit-area__iframe {
  background-color: $input-disabled-bg;
}
```
This style is applied to the `iframe` wrapper element inside of the `TinyMCE` editor to make the text field appear gray when the editor is in disabled state. I wasn't able to find a way to pass a class name to `TinyMCE` editor which it would apply to the `iframe` wrapper. I also wasn't able to find a way to replicate the same behavior without using the bootstrap variable or hard-coding the color value.

### Removed `.react-autosuggest__input` class

Were able to remove `.react-autosuggest__input` class, because it is not applied to the input element. Instead this element has class `.form-control`, because it's overwritten by `inputProps` in `src/components/ListField/index.jsx` on lines 183 and 236.

In other words, these styles are never used, so removing them.

## Private references

`Private-ref`: [BB-6217](https://tasks.opencraft.com/browse/BB-6217)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Cup0fCoffee!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@Cup0fCoffee Thank you for your contribution. Is this ready for our review?

**Comment by Cup0fCoffee** [comment]
@natabene Not yet, I'm waiting for @xitij2000 to review this PR first. I'll let you know once the PR is ready for review.

File: src/components/ListField/ListField.scss [hunk 1]
```diff
@@ -1,54 +1,15 @@
...
-  position: relative;
-}
-
-.react-autosuggest__input {
-  width: 100%;
-  height: 30px;
-  padding: 10px 20px;
-  border: 1px solid $gray-600;
-  border-radius: 0.1875rem;
-}
```
Comment (xitij2000) [inline src/components/ListField/ListField.scss:11]: I don't see this copied over.

File: src/components/ListField/index.jsx [hunk 2]
```diff
@@ -235,6 +235,12 @@ class ListField extends React.Component {
             onSuggestionSelected={this.onSuggestionEntered}
             alwaysRenderSuggestions={searchString.length > 2}
             id={`${itemType}-search`}
+            theme={{
+              container:                'react-autosuggest__container position-relative',
+              suggestionsContainerOpen: 'react-autosuggest__suggestions-container--open bg-white border-gray-300 w-100',
+              suggestionsList:          'react-autosuggest__suggestions-list m-0 p-0 list-unstyled overflow-auto',
+              suggestionHighlighted:    'react-autosuggest__suggestion--highlighted bg-gray-300',
+            }}
```
Comment (xitij2000) [inline src/components/ListField/index.jsx:243]: The styling here seems a bit off, I think the values here should be next to the keys, not aligned at the same column.

**Review by xitij2000 — APPROVED** [review]
:+1:  I just see a few small issues here, other than that, good to go!

- [x] I tested this: test on devstack
- [x] I read through the code

**Comment by xitij2000** [comment]
>This style is applied to the iframe wrapper element inside of the TinyMCE editor to make the text field appear gray when the editor is in disabled state. I wasn't able to find a way to pass a class name to TinyMCE editor which it would apply to the iframe wrapper. I also wasn't able to find a way to replicate the same behavior without using the bootstrap variable or hard-coding the color value.

@Cup0fCoffee Thats OK! This goes a good way to removing dependence, and for cases where we can't remove the dependence, I'm working on a PR to use CSS variables instead.

File: src/components/ListField/ListField.scss (hunk 1 above)
Comment (Cup0fCoffee) [reply src/components/ListField/ListField.scss:11]: I've provided explanation in the commit message 89f87af:
```
    refactor: remove unused styles for autosuggest input

    Removing these styles as part of the effor of reducing the use of
    bootstrap variables in SCSS. Were able to remove these styles
    completely, because the `.react-autosuggest__input` class is not applied
    to the input element. Instead this element has class `form-control`,
    because it's overwritten by `inputProps` in
    `src/components/ListField/index.jsx` on lines 183 and 236.

    In other words, these styles are never used, so removing them.
```

File: src/components/ListField/index.jsx (hunk 2 above)
Comment (Cup0fCoffee) [reply src/components/ListField/index.jsx]: Fixed in f7d0ffa.

File: src/components/ListField/ListField.scss (hunk 1 above)
Comment (Cup0fCoffee) [reply src/components/ListField/ListField.scss:11]: Added a note to the description about this.

**Comment by Cup0fCoffee** [comment]
@natabene This PR is ready for review.

**Comment by natabene** [comment]
I cannot trigger the re-run of the tests, and cannot close/open the PR to force the tests to run. Whoever is reviewing this, please help with running the tests.

**Review by ansabgillani — APPROVED** [review]

**Comment by natabene** [comment]
@ansabgillani Please be reminded that when the time comes, you will also need to merge since author cannot.

**Comment by Cup0fCoffee** [comment]
@ansabgillani Seems like I forgot to update the snapshots. I'm going to do it now, and push.

**Comment by ansabgillani** [comment]
Thanks @Cup0fCoffee, Updates to tests are mandatory.

**Comment by Cup0fCoffee** [comment]
@ansabgillani @AliAdnanSohail Sorry for that.
I've updated the snapshots and ran the tests locally to make sure they are all passing. The test should no longer fail.

**Comment by ansabgillani** [comment]
Thanks @Cup0fCoffee 
Really appreciate the effort.

**Review by AliAdnanSohail — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@Cup0fCoffee 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-publisher&entry.867055334=https://github.com/openedx/frontend-app-publisher/pull/774&entry.1484655318=https://github.com/Cup0fCoffee&entry.752974735=2022-08-26+14:38&entry.1917517419=2022-09-06+14:39&entry.2133058324=Yes) so we can improve your experience in the future.
